ufo2fontir/src/toir.rs

```diff
@@ -84,9 +67,44 @@ fn to_ir_glyph_instance(glyph: &norad::Glyph) -> ir::GlyphInstance {
     }
 }
 
+pub fn to_ir_axis(axis: &designspace::Axis) -> ir::Axis {
+    // <https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#axis-element>
+    let min = UserCoord::new(axis.minimum.unwrap().into());
+    let default = UserCoord::new(axis.default.into());
+    let max = UserCoord::new(axis.maximum.unwrap().into());
+
+    // <https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#map-element>
+    let converter = if let Some(mappings) = &axis.map {
+        let examples: Vec<_> = mappings
+            .iter()
+            .map(|map| {
+                (
+                    UserCoord::new(OrderedFloat(map.input)),
+                    DesignCoord::new(OrderedFloat(map.output)),
+                )
+            })
+            .collect();
+        let default_idx = examples.iter().position(|(u, _)| *u == default).expect(
+            "We currently require that you have a mapping for the default if you have mappings",
```

Comment:
to build an avar, fonttools actually requires not only a mapping for the default axis coordinate, but also for the axis minimum and maximum as well. Those are encoded directly in the fvar.
Simply taking the miniumum/maximum of the provided (potentionally incomplete) axis mappings is not sufficient, because what makes a user-space coordinate minimum or maximum (and hence -1 and +1 in normalized coordinates) is the fact that it is globally specificed as such in the fvar table. It's better to enforce this here.

Reply:
That's an interesting limitation. I don't actually find it obvious that should be required. I'm going to put in the validation and file an issue.

Reply:
On a moments further thought the sort of "guessing" at the intended design range that would be implied seems ugly, forget that.